commands/deploy.go

```diff
@@ -343,9 +343,42 @@ func (p *deploycmd) deployFuncV20180708(c *cli.Context, app *models.App, funcfil
 	if funcfile.Name == "" {
 		funcfile.Name = filepath.Base(filepath.Dir(funcfilePath)) // todo: should probably make a copy of ff before changing it
 	}
-	fmt.Printf("Deploying %s to app: %s\n", funcfile.Name, app.Name)
 
-	var err error
+	oracleProvider, _ := getOracleProvider()
+	if oracleProvider != nil && oracleProvider.ImageCompartmentID != "" {
+		// If the provider is Oracle and ImageCompartmentID is present, we need to deploy image to the ImageCompartmentID.
+
+		// Check Registry and build the repository name.
```

Comment:
nit: most of these small comments aren't doing a lot for me. e.g., here "check registry and build the repository name" just says exactly what the code is doing. Personally I think it would be better (but not required) to write a small comment at the top describing roughly /why/ things are happening the way they are. e.g., "A repository must have a unique name within the tenancy. If one doesn't exist create it in the compartment; if one exists already check to see if it is in the right compartment. If it's not in the right compartment we can't continue".

Reply:
Here the comments I am leaving are like steps and the code is what the step does. Its much easier to understand what is happening later on when reading than to see a chunk of code and figure out what it does, especially for someone new. I'll add why we do it at the top and space it so it helps in a similar way.